[BUG] Handle multiple `UserMessage` text contents in `DefaultAiService`

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->

I created a `CustomContentIngector` that puts multiple text content elements in `UserMessage`, however an exception was thrown from the `appendOutputFormatInstructions()` method in `DefaultAiService` while calling `userMessage.singleText()` assuming it's a single text message.

I wonder is this expected?

**Log and Stack trace**
<!-- Please provide a log and a stack trace (with exception), if applicable. -->

```
Caused by: java.util.UnknownFormatConversionException: Conversion = 'r'
	at java.util.Formatter$FormatSpecifier.conversion(Formatter.java:2958)
	at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2994)
	at java.util.Formatter.parse(Formatter.java:2849)
	at java.util.Formatter.format(Formatter.java:2774)
	at java.util.Formatter.format(Formatter.java:2728)
	at java.lang.String.formatted(String.java:4452)
	at dev.langchain4j.internal.Exceptions.runtime(Exceptions.java:36)
	at dev.langchain4j.data.message.UserMessage.singleText(UserMessage.java:124)
	at dev.langchain4j.service.DefaultAiServices$1.appendOutputFormatInstructions(DefaultAiServices.java:335)
	at dev.langchain4j.service.DefaultAiServices$1.invoke(DefaultAiServices.java:207)
	at com.canva.creatorcontentpublish.server.services.strategist.$Proxy109.chat(Unknown Source)
	at com.canva.creatorcontentpublish.server.services.strategist.StrategistService.chat(StrategistService.java:198)
```

**To Reproduce**
<!-- Please provide a relevant code snippets to reproduce this bug. -->

Create a `CustomContentIngector` that returns a `UserMessage` with multiple text content elements. Create an `OpenAiOfficialChatModel` without `RESPONSE_FORMAT_JSON_SCHEMA` capability support, and no streaming enabled.

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

**Please complete the following information:**
- LangChain4j version: `v1.2.0`
- LLM(s) used: `GPT-4o-mini`
- Java version: `21`
- Spring Boot version (if applicable): 

**Additional context**
<!-- Add any other context about the problem here. -->

I'm assuming it could be sth like e.g.

```
private UserMessage appendOutputFormatInstructions(Type returnType, UserMessage userMessage) {
  String outputFormatInstructions = serviceOutputParser.outputFormatInstructions(returnType);
  String text = userMessage.getFirstTextContent() + outputFormatInstructions; // <--- replacing userMessage.singleText()
```

--- a/langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java
+++ b/langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java
@@ -6,9 +6,9 @@ import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
 import static dev.langchain4j.model.chat.Capability.RESPONSE_FORMAT_JSON_SCHEMA;
 import static dev.langchain4j.model.chat.request.ResponseFormatType.JSON;
 import static dev.langchain4j.model.output.FinishReason.TOOL_EXECUTION;
+import static dev.langchain4j.service.AiServiceValidation.validateParameters;
 import static dev.langchain4j.service.IllegalConfigurationException.illegalConfiguration;
 import static dev.langchain4j.service.TypeUtils.typeHasRawClass;
-import static dev.langchain4j.service.AiServiceValidation.validateParameters;
 import static dev.langchain4j.spi.ServiceHelper.loadFactories;
 
 import dev.langchain4j.Internal;
@@ -416,10 +416,23 @@ class DefaultAiServices<T> extends AiServices<T> {
                             return userMessage;
                         }
 
-                        String newText = userMessage.singleText() + outputFormatInstructions;
-                        return userMessage.toBuilder()
-                                .contents(List.of(TextContent.from(newText)))
-                                .build();
+                        List<Content> contents = new ArrayList<>(userMessage.contents());
+
+                        boolean appended = false;
+                        for (int i = contents.size() - 1; i >= 0; i--) {
+                            if (contents.get(i) instanceof TextContent lastTextContent) {
+                                String newText = lastTextContent.text() + outputFormatInstructions;
+                                contents.set(i, TextContent.from(newText));
+                                appended = true;
+                                break;
+                            }
+                        }
+
+                        if (!appended) {
+                            contents.add(TextContent.from(outputFormatInstructions));
+                        }
+
+                        return userMessage.toBuilder().contents(contents).build();
                     }
 
                     private Future<Moderation> triggerModerationIfNeeded(Method method, List<ChatMessage> messages) {

--- a/langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithCustomContentInjectorTest.java
+++ b/langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithCustomContentInjectorTest.java
@@ -1,0 +1,145 @@
+package dev.langchain4j.service;
+
+import static dev.langchain4j.service.AiServicesIT.verifyNoMoreInteractionsFor;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+
+import dev.langchain4j.data.image.Image;
+import dev.langchain4j.data.message.ImageContent;
+import dev.langchain4j.data.message.TextContent;
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.model.chat.mock.ChatModelMock;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.rag.DefaultRetrievalAugmentor;
+import dev.langchain4j.rag.RetrievalAugmentor;
+import dev.langchain4j.rag.content.injector.ContentInjector;
+import dev.langchain4j.rag.content.retriever.ContentRetriever;
+import java.util.Collections;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class AiServicesWithCustomContentInjectorTest {
+
+    @Spy
+    ChatModel chatModel = ChatModelMock.thatAlwaysResponds("{\"value\":\"test\"}");
+
+    @AfterEach
+    void afterEach() {
+        verifyNoMoreInteractionsFor(chatModel);
+    }
+
+    private static final Image image = Image.builder()
+            .url("https://en.wikipedia.org/wiki/Llama#/media/File:Llamas,_Vernagt-Stausee,_Italy.jpg")
+            .build();
+
+    private static final ImageContent imageContent = ImageContent.from(image);
+
+    static class MyStructuredOutput {
+
+        private String value;
+
+        MyStructuredOutput() {}
+
+        MyStructuredOutput(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        public void setValue(String value) {
+            this.value = value;
+        }
+    }
+
+    interface MyAiService {
+        MyStructuredOutput chat(@UserMessage String text);
+    }
+
+    @Test
+    void should_append_output_format_instructions_to_the_last_text_content() {
+        // given
+        ContentInjector customContentInjector = (contents, chatMessage) -> {
+            return dev.langchain4j.data.message.UserMessage.from(
+                    TextContent.from("First text"), imageContent, TextContent.from("Second text"));
+        };
+
+        ContentRetriever customContentRetriever = query -> Collections.emptyList();
+
+        RetrievalAugmentor retrievalAugmentor = DefaultRetrievalAugmentor.builder()
+                .contentRetriever(customContentRetriever)
+                .contentInjector(customContentInjector)
+                .build();
+
+        MyAiService myAiService = AiServices.builder(MyAiService.class)
+                .chatModel(chatModel)
+                .retrievalAugmentor(retrievalAugmentor)
+                .build();
+
+        // when
+        myAiService.chat("How many lamas are there in this image?");
+        ArgumentCaptor<ChatRequest> chatRequestCaptor = ArgumentCaptor.forClass(ChatRequest.class);
+        verify(chatModel).chat(chatRequestCaptor.capture());
+
+        dev.langchain4j.data.message.UserMessage userMessage = (dev.langchain4j.data.message.UserMessage)
+                chatRequestCaptor.getValue().messages().get(0);
+
+        assertThat(userMessage.contents()).hasSize(3);
+        assertThat(userMessage.contents().get(0)).isInstanceOf(TextContent.class);
+        assertThat(userMessage.contents().get(1)).isInstanceOf(ImageContent.class);
+        assertThat(userMessage.contents().get(2)).isInstanceOf(TextContent.class);
+
+        TextContent text1 = (TextContent) userMessage.contents().get(0);
+        TextContent text2 = (TextContent) userMessage.contents().get(2);
+        assertThat(text1.text()).isEqualTo("First text");
+        assertThat(text2.text())
+                .contains("Second text")
+                .contains("You must answer strictly in the following JSON format");
+
+        verify(chatModel).supportedCapabilities();
+    }
+
+    @Test
+    void should_create_new_text_content_of_output_format_instructions_if_there_is_no_text_content() {
+        // given
+        ContentInjector customContentInjector = (contents, chatMessage) -> {
+            return dev.langchain4j.data.message.UserMessage.from(imageContent, imageContent);
+        };
+
+        ContentRetriever customContentRetriever = query -> Collections.emptyList();
+
+        RetrievalAugmentor retrievalAugmentor = DefaultRetrievalAugmentor.builder()
+                .contentRetriever(customContentRetriever)
+                .contentInjector(customContentInjector)
+                .build();
+
+        MyAiService myAiService = AiServices.builder(MyAiService.class)
+                .chatModel(chatModel)
+                .retrievalAugmentor(retrievalAugmentor)
+                .build();
+
+        // when
+        myAiService.chat("How many lamas are there in this image?");
+        ArgumentCaptor<ChatRequest> chatRequestCaptor = ArgumentCaptor.forClass(ChatRequest.class);
+        verify(chatModel).chat(chatRequestCaptor.capture());
+
+        dev.langchain4j.data.message.UserMessage userMessage = (dev.langchain4j.data.message.UserMessage)
+                chatRequestCaptor.getValue().messages().get(0);
+
+        assertThat(userMessage.contents()).hasSize(3);
+        assertThat(userMessage.contents().get(0)).isInstanceOf(ImageContent.class);
+        assertThat(userMessage.contents().get(1)).isInstanceOf(ImageContent.class);
+        assertThat(userMessage.contents().get(2)).isInstanceOf(TextContent.class);
+
+        TextContent textContent = (TextContent) userMessage.contents().get(2);
+        assertThat(textContent.text()).contains("You must answer strictly in the following JSON format");
+
+        verify(chatModel).supportedCapabilities();
+    }
+}

